daemon/api/dependencies.py

```diff
@@ -46,13 +80,20 @@ def localpath(self) -> Path:
     @cached_property
     def port_expose(self) -> str:
         """
-        Sets `port_expose` for the Flow started in `mini-jinad`.
-        NOTE: this port needs to be exposed before starting `mini-jinad`, hence set here.
+        Sets `port_expose` for the Flow started in `partial-daemon`.
+        This port needs to be exposed before starting `partial-daemon`, hence set here.
+        If env vars are passed, set them in current env, to make sure all values are replaced.
+        Before loading the flow yaml, change CWD to workspace dir.
 
         :return: port_expose
         """
-        f = Flow.load_config(str(self.localpath()))
-        return f.port_expose or random_port()
+        with ExitStack() as stack:
+            if self.envs:
+                for key, val in self.envs.items():
+                    stack.enter_context(change_env(key, val))
```

Comment:
This is changing env variables back and forth on the jinad host and not in the child docker container right? Isnt that dangerous as it could happen concurrently and messes things up?

Reply:
Good point. This is changing the env on the main jinad process. We can possibly add an `asyncio.Lock` to avoid other events conflicting. I would do it in a different PR though, as adding this lock via `async..await` needs to be propagated to the endpoint level.